File: DataBinding/Section/IListBinding/C#/IListBinding.DataLayer/DataProvider.cs
using System.Data.OleDb;

namespace GrapeCity.ActiveReports.Samples.IListBinding.DataLayer
{
	internal class DataProvider
	{
		/// <summary>
		/// Returns a new connection object for reading the data in the ProductCollection
		/// </summary>
		internal static OleDbConnection NewConnection
		{
			get
			{
				return new OleDbConnection(GrapeCity.ActiveReports.IListBinding.DataLayer.Properties.Resources.ConnectionString);
			}
		}
	}
}

File: DataBinding/Section/IListBinding/VB.NET/IListBinding.DataLayer/DataProvider.vb
Imports System.Data.OleDb

Friend Class DataProvider

	'<summary>
	' Returns a new connection object for reading the data in the ProductCollection
	'</summary>
	Friend Shared ReadOnly Property NewConnection() As OleDbConnection
		Get
			Dim connectString As String = My.Resources.ConnectionString
			Return New OleDbConnection(connectString)
		End Get
	End Property

End Class 'DataProvider
